e2e/wdio/core/tests/calendar.e2e-spec.ts

```diff
@@ -226,11 +229,20 @@ describe('calendar test suite', () => {
     describe('calendar with special days example', () => {
         it('should check ability to mark weekends', () => {
             click(specialDaysCalendar + calendarOptions);
-            expect(doesItExist(specialDaysCalendar + markedDays)).toBe(true);
-            click(specialDaysCalendar + calendarOptions);
-            expect(doesItExist(specialDaysCalendar + markedDays)).toBe(false);
+            for (let i = 0; i < getElementArrayLength(specialDaysCalendar + weekendDays); i++) {
+                expect(getElementClass(specialDaysCalendar + weekendDays, i)).toContain('special-day');
+            }
+            // need to additionality check first and last days in calendar cz if they are days of other month
+            // they do not have class 'weekend', but anyway will be marked as weekend
+            const itemsLength = getElementArrayLength(specialDaysCalendar + calendarItem);
+            if (getElementClass(specialDaysCalendar + calendarItem).includes('other-month')) {
+                expect(getElementClass(specialDaysCalendar + calendarItem)).toContain('special-day');
+            }
+            if (getElementClass(specialDaysCalendar + calendarItem, itemsLength - 2).includes('other-month')) {
+                expect(getElementClass(specialDaysCalendar + calendarItem, itemsLength - 2)).toContain('special-day');
+            }
```

Comment:
can we avoid ifs inside test?
otherwise, we are getting different expections and as an outcome different test results.
I believe during testing we must do assertion without such branching